lib/set-archive-author.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+// Find a username specified in the option "archive_author"
+// Set the post_author (in $data) to be the id corresponding to the user
+
+function set_archive_author($data, $postarr)
+{
+	$fix_types = ["page", "post", "attachment"];
+
+	if (!in_array($data['post_type'], $fix_types)) {
+		return $data;
+	}
+
+	$archive_username = get_option('archive_author');
+
+	if (!empty($archive_author)) {
+		if (is_int($archive_author)) {
+			$archive_author_id = $archive_author;
+		} else {
+			$archive_author_id = get_user_by('id', $archive_author);
```

Comment:
Bug: `get_user_by` returns a `WP_User` object. If we're assuming that the `archive_author` option value will always be an integer, even if it's returned as a string, do we need to do anything here other than maybe convert it to the integer type? e.g.

`$archive_author_id = intval($archive_author)`

`get_option` will often return a string even if the value in the database is an integer, so we can't assume that the fact the `is_int()` test has failed indicates that we should assume the value is a text string, rather than an integer in string format.